napari/_qt/widgets/qt_dims_slider.py

```diff
@@ -610,12 +627,19 @@ def work(self):
                 self.frame_requested.emit(self.axis, self.min_point)
             elif self.step < 0 and self.current <= self.min_point + 1:
                 self.frame_requested.emit(self.axis, self.max_point)
-            self.timer.singleShot(int(self.interval), self.advance)
+            self.timer.start()
         else:
             # immediately advance one frame
             self.advance()
         self.started.emit()
 
+    @ensure_object_thread
```

Comment:
I got some warnings when closing napari: 

```
WARNING: QObject::killTimer: Timers cannot be stopped from another thread
11:00:36 WARNING QObject::killTimer: Timers cannot be stopped from another thread
WARNING: QObject::~QObject: Timers cannot be stopped from another thread
11:00:36 WARNING QObject::~QObject: Timers cannot be stopped from another thread
```

That might be related to the test failures?

Maybe we need to pass this as a parent to the `QTimer`?

Reply:
I have found that I terminate the worker and thread in the same place. It means that the termination event is not processed as the thread is ended. 

I solve it by move the worker again to the main thread and then terminating.

Reply:
That fixed the tests, but I still get the error message when using playback on a 3D image (displayed in 2D), then closing napari.

https://user-images.githubusercontent.com/2608297/206309081-c55a9545-44fd-4ca6-be89-49304c9fddef.mp4

Reply:
Ah. I fix only showing error when user click stop button (it happens sometimes). I will fix it also.

Reply:
@andy-sweet could you check now?

Reply:
Works now! I have some minor (hopefully!) follow-up questions on the changes that made this work. Thanks again for your work on this and for being patient with my questions!